crates/bevy_utils/src/lib.rs

```diff
@@ -340,38 +340,39 @@ pub type EntityHashSet<T> = hashbrown::HashSet<T, EntityHash>;
 
 /// A specialized hashmap type with Key of [`TypeId`]
 /// Iteration order only depends on the order of insertions and deletions.
-pub type TypeIdMap<V> = hashbrown::HashMap<TypeId, V, NoOpTypeIdHash>;
+pub type TypeIdMap<V> = hashbrown::HashMap<TypeId, V, NoOpHash>;
 
-/// [`BuildHasher`] for [`TypeId`]s.
-#[derive(Default)]
-pub struct NoOpTypeIdHash;
+/// [`BuildHasher`] for types that already contain a high-quality hash.
+#[derive(Clone, Default)]
+pub struct NoOpHash;
```

Comment:
This is technically a breaking change.

Reply:
Do you want me to do something like adding it to the release notes, etc? Most of this patch is a breaking change, really.

Reply:
Before this PR is merged, please add a `Migration Guide` section to your PR description.